### ISSUE TITLE
Times.Never is incorrectly reported in a setup with Times.Exactly

### ISSUE THREAD

**Issue by kkm000** [moq/moq4#211]
The following calll to `Verify`

```
      mock.Verify(x => x.Create(It.IsAny<CspParameters>()), Times.Exactly(11);
```

results in this message:

```
Moq.MockException : 
Expected invocation on the mock exactly 11 times, but was 1 times: x => x.Create(It.IsAny<CspParameters>())

Configured setups:
x => x.Create(It.IsAny<CspParameters>()), Times.Never

Performed invocations:
....
```

Note misreported `Times.Never`. Expected in the message either `Times.Exactly(11)`, or a message to the effect of absence of any configured setups, because there were truly none. The number is reported correctly in the preceding text however.

**Comment by LeonidLevin** [comment]
I cannot reproduce this issue. This is my test: 

```
			[Fact]
			public void IfNoSetupsAndTestFails_ReportNoConfiguredSetups()
			{
				var mock = new Mock<ISimpleInterface>();

				var a = new MyClass();

				mock.Object.Method(a);

				mock.Verify(x => x.Method(It.IsAny<MyClass>()), Times.Exactly(11));
			}

			public interface ISimpleInterface
			{
				int Method(MyClass a);
			}

			public class MyClass { }
```

The test results are as follows:


```
Moq.MockException : 
Expected invocation on the mock exactly 11 times, but was 1 times: x => x.Method(It.IsAny<MyClass>())
No setups configured.
```

The "No setups configured" message appears as there were no setups.

**Comment by kkm000** [comment]
Has a new version been out since? This bug turns 7 months tomorrow...

**Comment by kzu** [comment]
You bet :p

**Comment by stakx** [comment]
@kkm000, it's been a while, but if you are still willing to help diagnose this error, could you please provide a small but complete program or unit test that reproduces the problem?

(I might be closing this issue in about 3 weeks' time if you don't respond before that, but we can always reopen the issue if it just takes you a little longer to reply. ;-)

**Comment by kkm000** [comment]
@stakx: Let me try tomorrow, thanks! It's been a while, so the bug might have fixed itself since then. I'll test.

**Comment by stakx** [comment]
This could be fixed (the bug sits in `Mock.FormatCallCount` called from `Mock.FormatSetupsInfo`), but whether it's actually worth fixing it depends on the way #373:

 * The ability to specify the expected number of calls during setup has been deprecated long ago, so perhaps that bit of information should probably also be removed from the exception message.

 * However, if the ability to specify `Times` during setup makes it back into Moq, then this error here needs to be fixed as well.

**Comment by stakx** [comment]
Here's a repro for this issue:

```csharp
var mock = new Mock<IX>(MockBehavior.Never);
mock.Setup(x => x.Create(It.IsAny<CspParameters>()));
mock.Object.Create(null);
mock.Object.Create(null);
mock.Verify(x => x.Create(It.IsAny<CspParameters>()), Times.Exactly(11));

public interface IX
{
    void Create(CspParameters parameters);
}

public class CspParameters { }
```

Which produces the following exception message:

```
Unhandled Exception: Moq.MockException:
Expected invocation on the mock exactly 11 times, but was 2 times: x => x.Create
(It.IsAny<CspParameters>())

Configured setups:
x => x.Create(It.IsAny<CspParameters>()), Times.Exactly(2)

Performed invocations:
IX.Create(null)
IX.Create(null)
   at Moq.Mock.ThrowVerifyException(MethodCall expected, IEnumerable`1 setups, IEnumerable`1 actualCalls, Expression expression, Times times, Int32 callCount)
   at Moq.Mock.VerifyCalls(Interceptor targetInterceptor, MethodCall expected, Expression expression, Times times)
   at Moq.Mock.Verify[T](Mock`1 mock, Expression`1 expression, Times times, String failMessage)
   at Moq.Mock`1.Verify(Expression`1 expression, Times times)
```

Note how it misreports the number of actual invocations as the number of expected invocations,

**Comment by stakx** [comment]
This should be fixed with the next release of Moq (version >4.7.63).
